### PR TITLE
mongodb API route

### PR DESCRIPTION
**Pull request by Nirb8**

### CODE REVIEW

**Comment by Nirb8** [comment]
Added support for zip file download as well, still awaiting code cleanup and suggestions from @budak7273

File: apps/frontend/lib/mongodb.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// THIS IS CURRENTLY UNUSED, FIGURE OUT HOW TO IMPORT IT INTO api/experiments/
```
Comment (budak7273) [inline apps/frontend/lib/mongodb.ts:1]: Remove this file?

File: apps/frontend/firebase/db.ts [hunk 2]
```diff
@@ -74,26 +74,34 @@ const downloadArbitraryFile = (url: string, name: string) => {
...
+		method: 'POST',
+		headers: new Headers({
+			'Content-Type': 'application/json',
+		}),
+		body: JSON.stringify(expId),
 	});
+	const record = await response.json();
+	const csvContents = record[0].resultContent;
+	const url = `data:text/plain;charset=utf-8,${encodeURIComponent(csvContents)}`;
+	downloadArbitraryFile(url, `result${expId}.csv`);
```
Comment (budak7273) [inline apps/frontend/firebase/db.ts:89]: See my comments on the api routes

File: apps/frontend/pages/api/experiments/csv.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { MongoClient } from 'mongodb';
+
+const mongoCSVHandler = async (req, res) => {
```
Comment (budak7273) [inline apps/frontend/pages/api/experiments/csv.tsx:3]: Switch to follow the pattern that the `[id].tsx` file follows for consistency + ability to copy a download link that doesn't need request body

So this file would move to /pages/api/downloads/csv/[id].tsx` for example

That way a "real link" would be `http://glados.csse.rose-hulman.edu/api/downloads/csv/experimentIdGoesHere`, and you could go to a download link without needing to use request body details to specify what experiment to download

File: apps/frontend/pages/api/experiments/csv.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	try {
+		const key = req.body;
+		console.log('Key from request body is', key);
+
+		const uri = process.env.MONGODB_URI;
+		let clientPromise : Promise<MongoClient> = new Promise((success) => {
+			return true;
+		});
+		clientPromise = new MongoClient('mongodb://glados-mongodb:27017', {}).connect();
```
Comment (budak7273) [inline apps/frontend/pages/api/experiments/csv.tsx:12]: Move this out to an env file or constant somewhere

File: apps/frontend/pages/api/experiments/zip.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+		});
+		clientPromise = new MongoClient('mongodb://glados-mongodb:27017', {}).connect();
+		const client = await clientPromise;
+		const db = client.db("gladosdb");
+
+		const resultZip = await db
+			.collection('zips')
+			.find({'_id': key}).toArray();
+		console.log('found results: ');
+		res.json(resultZip);
```
Comment (budak7273) [inline apps/frontend/pages/api/experiments/zip.tsx:20]: Code duplication with the csv route, the only thing that changes is the collections 'results' to 'zips'. Put in a library somewhere?

**Review by budak7273 — DISMISSED** [review]

**Review by budak7273 — APPROVED** [review]

**Comment by budak7273** [comment]
David said it's good to merge on Discord

**Pull request merged**